### PR TITLE
[SDL3] Cleanup void functions

### PR DESCRIPTION
**Pull request by LiquidityC**
Some functions that do call SDL_SetError but return void changed to instead return non-zero in case of errors.
This commit doesn't handle all such cases. There is more to do.

If more of these PRs are of interest I can keep looking through the files. Perhaps some files need more attention then others? (please point to priority files in that case).
If you prefer larger PRs I can expand this one.

There might still be more cases of these kinds of functions in SDL_video but this is my first foray into "this side" of SDL so I figure I'd rather start small. Rather then submitting walls of code.

## Description
Looked through the functions in SDL_video.h and SDL_video.c and changed relevant functions that can call `SDL_SetError()` to return a non-zero response in such cases rather then just returning void.

**The following functions have been updated:**

- `SDL_SetWindowTitle`
- `SDL_SetWindowIcon`
- `SDL_SetWindowSize`
- `SDL_SetWindowMinimumSize`
- `SDL_SetWindowMaximumSize`

## Existing Issue(s)
This relates to #7233 but doesn't complete it. I'm certain there are more functions to address but as mentioned above I'd rather start small.

### CODE REVIEW

File: src/video/SDL_video.c [hunk 1]
```diff
@@ -2402,22 +2405,22 @@ void SDL_GetWindowSizeInPixels(SDL_Window *window, int *w, int *h)
...
 }
 
-void SDL_SetWindowMinimumSize(SDL_Window *window, int min_w, int min_h)
+int SDL_SetWindowMinimumSize(SDL_Window *window, int min_w, int min_h)
 {
-    CHECK_WINDOW_MAGIC(window, );
+    CHECK_WINDOW_MAGIC(window, -1);
     if (min_w <= 0) {
         SDL_InvalidParamError("min_w");
-        return;
```
Comment (slouken) [inline src/video/SDL_video.c:2410]: and here (etc.)

File: src/video/SDL_video.c [hunk 2]
```diff
@@ -2284,16 +2286,16 @@ void SDL_SetWindowAlwaysOnTop(SDL_Window *window, SDL_bool on_top)
...
 }
 
-void SDL_SetWindowSize(SDL_Window *window, int w, int h)
+int SDL_SetWindowSize(SDL_Window *window, int w, int h)
 {
-    CHECK_WINDOW_MAGIC(window, );
+    CHECK_WINDOW_MAGIC(window, -1);
     if (w <= 0) {
         SDL_InvalidParamError("w");
-        return;
```
[the remark this answers is not in the document]
Comment (LiquidityC) [reply src/video/SDL_video.c:2292]: Of course :man_facepalming:

File: src/video/SDL_video.c (hunk 1 above)
Comment (LiquidityC) [reply src/video/SDL_video.c:2410]: Fixed

File: src/video/SDL_video.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (LiquidityC) [reply src/video/SDL_video.c:2292]: Fixed all occurences I could find/think of

**Comment by slouken** [comment]
Looks good, thanks!

**Pull request merged**